libbpf-tools/trace_helpers.c

```diff
@@ -395,3 +398,112 @@ bool is_kernel_module(const char *name)
 	fclose(f);
 	return found;
 }
+
+bool fentry_exists(const char *name, const char *mod)
+{
+	const char sysfs_vmlinux[] = "/sys/kernel/btf/vmlinux";
+	struct btf *base, *btf = NULL;
+	const struct btf_type *type;
+	const struct btf_enum *e;
+	char sysfs_mod[64];
+	int id = -1, i;
+
+	base = btf__parse(sysfs_vmlinux, NULL);
+	if (libbpf_get_error(base)) {
+		fprintf(stderr, "failed to parse vmlinux BTF at '%s': %s\n",
+			sysfs_vmlinux, strerror(-libbpf_get_error(base)));
+		base = NULL;
+		goto err_out;
+	}
+	if (mod) {
+		snprintf(sysfs_mod, sizeof(sysfs_mod), "/sys/kernel/btf/%s", mod);
+		btf = btf__parse_split(sysfs_mod, base);
+		if (libbpf_get_error(btf)) {
+			fprintf(stderr, "failed to load BTF from %s: %s\n",
+				sysfs_mod, strerror(-libbpf_get_error(btf)));
+			btf = NULL;
+			goto err_out;
+		}
+	} else {
+		btf = base;
+	}
+
+	id = btf__find_by_name_kind(btf, "bpf_attach_type", BTF_KIND_ENUM);
+	if (id < 0)
+		goto err_out;
+	type = btf__type_by_id(btf, id);
+	if (!type) {
+		id = -1;
+		goto err_out;
+	}
```

Comment:
this can't happen, if btf__find_by_name_kind() returned ID, it will be valid, so please drop this check altogether